.github/workflows/quality.yml

```diff
@@ -0,0 +1,32 @@
+name: Quality
+
+on:
+  push:
+    branches:
+      - dev
+  pull_request:
+    branches:
+      - dev
+
+jobs:
+  eslint:
```

Comment:
@pettinarip realizing that there might be some trouble with this job when triggered on a pull request.

Because of Github security limitations with forks, it might not be possible to run a commit from a file change done by this job.

See the `git-auto-commit-action` [README section regarding public repos with forks](https://github.com/stefanzweifel/git-auto-commit-action?tab=readme-ov-file#use-in-forks-from-public-repositories)

There should be no issue running this job on push to the base branch (`dev`).

So with that said, I'm thinking of two parts here:

1. Job that just runs on pull request to simply execute `yarn lint` and returns any errors/warnings
2. Job that runs on push to base branch and executes `yarn lint:fix` to auto fix any warnings or errors that can be done and then make a commit to push separately.

Though it's starting to sound like running a pre-commit is easier here! 😅 

Is there something I am missing that you might know about?

Reply:
Ah good catch! yea, I think that at this point, a pre commit hook would be just easier. We could use something like https://typicode.github.io/husky/ that is a well-known lib for this.

Reply:
@pettinarip, I've used Husky. So I will look at converting to the pre-commit! 😁

Reply:
Yeah, that's what I used to use for this sort of thing too. I don't know if there is a better new thing now, but Husky should work fine.

Reply:
@pettinarip not that I know of. Just combo Husky with `lint-staged`

Reply:
Exactly. Ok.